DigitalLearningSolutions.Web/ViewModels/TrackingSystem/Delegates/GroupCourses/AddCourseToGroupCoursesViewModel.cs

```diff
@@ -6,32 +6,56 @@
     using DigitalLearningSolutions.Web.Helpers;
     using DigitalLearningSolutions.Web.ViewModels.Common.SearchablePage;
 
-    public class AddCourseToGroupCoursesViewModel
+    public class AddCourseToGroupCoursesViewModel : BaseSearchablePageViewModel
     {
         public AddCourseToGroupCoursesViewModel(
             IEnumerable<CourseAssessmentDetails> courses,
+            IEnumerable<string> categories,
+            IEnumerable<string> topics,
+            int? adminCategoryFilter,
             int groupId,
-            string groupName
-        )
+            string groupName,
+            string? searchString,
+            string? filterBy,
+            int page
+        ) : base(searchString, page, true, filterBy: filterBy, searchLabel: "Search courses")
         {
             GroupId = groupId;
             GroupName = groupName;
+            AdminCategoryFilter = adminCategoryFilter;
 
+            var courseList = courses.ToList();
+            var searchedCourses = GenericSearchHelper.SearchItems(courseList, SearchString);
+            var filteredCourses = FilteringHelper.FilterItems(searchedCourses.AsQueryable(), filterBy).ToList();
             var sortedCourses = GenericSortingHelper.SortAllItems(
-                courses.AsQueryable(),
+                filteredCourses.AsQueryable(),
                 nameof(CourseAssessmentDetails.CourseName),
-                BaseSearchablePageViewModel.Ascending
-            );
-
-            var coursesToShow = sortedCourses.Take(10);
+                Ascending
+            ).ToList();
+            MatchingSearchResults = sortedCourses.Count;

```

Comment:
There is a method `SortFilterAndPaginate` from the base class `BaseSearchablePageViewModel` that allows us to remove some of these lines and simplify the class. Can this be used here?